code/modules/mob/new_player/new_player.dm

```diff
@@ -170,7 +170,19 @@
 
 		var/datum/species/S = all_species[client.prefs.species]
 		if(!check_species_allowed(S))
-			return 0
+			return FALSE
+
+		var/should_warn = TRUE
+		if(client.prefs.job_high == job.title)
+			should_warn = FALSE
+		else if(job.title in client.prefs.job_medium)
+			should_warn = FALSE
+		else if(job.title in client.prefs.job_low)
+			should_warn = FALSE
+
+		if(should_warn)
+			if(alert(client, "Are you sure you want to join as [job.title]?", "Confirm Job Selection", "Yes", "No") == "No")
```

Comment:
You might want to change the message here to specify why it's throwing the warning (i.e. "You don't have any preferences as [job.title]. Are you sure you want to join?" or something like that.) Otherwise, the code's fine. There's an argument to be made for putting the if statements in an `||`, but that's semantics.

Reply:
Fair points. I don't want to put them all in the same line since that can get messy, and harder to read.